addons/base_import/models.py

```diff
@@ -416,6 +537,68 @@ def _convert_import_data(self, record, fields, options, context=None):
+        value = value.strip()
+        float_regex = re.compile(r'([0-9.,]+)')
+        split_value = filter(None, float_regex.split(value))
+        if len(split_value) > 2:
+            # This is probably not a float
+            return False
+        if len(split_value) == 1:
+            if float_regex.search(split_value[0]) is not None:
+                return split_value[0]
+            return False 
+        else:
+            # String has been split in 2, locate which index contains the float and which does not
+            currency_index = 0
+            if float_regex.search(split_value[0]) is not None:
+                currency_index = 1
+            # Check that currency exists
+            currency = self.pool.get('res.currency').search(cr, uid, [('symbol', '=', split_value[currency_index])])
+            if len(currency):
+                return split_value[currency_index+1%2]
+            # Otherwise it is not a float with a currency symbol
+            return False
+
+    def _parse_float_from_data(self, cr, uid, data, index, name, options, context=None):
+        remove_parenthesis = remove_sign = False
+        thousand_separator = options.get('float_thousand_separator', ' ')
+        decimal_separator = options.get('float_decimal_separator', '.')
+        for line in data:
+            if line[index] == '':
+                continue
+            remove_parenthesis = False
+            if line[index].startswith('('):
+                remove_parenthesis = True
+            line[index] = line[index].replace(thousand_separator, '').replace(decimal_separator, '.')
+            line[index] = line[index].replace('(','').replace(')', '')
+            old_value = line[index]
+            line[index] = self._remove_currency_symbol(cr, uid, line[index])
+            if line[index] == False:
+                raise ValueError(_("Column %s contains incorrect values (value: %s)" % (name, old_value)))
+            if remove_parenthesis:
+                line[index] = -float(line[index])
+
+    def _parse_import_data(self, cr, uid, data, import_fields, record, options, context=None):
+        # Get fields of type date/datetime
+        all_fields = self.pool[record.res_model].fields_get(cr, uid, context=context)
+        for name, field in all_fields.iteritems():
+            if field['type'] in ('date', 'datetime') and name in import_fields:
+                # Parse date
+                index = import_fields.index(name)
+                dt = datetime.datetime
+                field_date_format = DEFAULT_SERVER_DATE_FORMAT if field['type'] == 'date' else DEFAULT_SERVER_DATETIME_FORMAT
+                if options.get('date_format', field_date_format) != field_date_format:
+                    for line in data:
+                        if line[index] != '':
+                            line[index] = dt.strftime(dt.strptime(line[index], options.get('date_format', field_date_format)), field_date_format)
```

Comment:
dont' we already know there is a `date_format` and it's different than `field_date_format`?